python/graphstorm/config/argument.py

```diff
@@ -221,10 +223,46 @@ def __init__(self, cmd_args):
         else:
             self._multi_tasks = None
 
-        # If model output is configured, save the updated config as a yaml file there
+        # If model output is configured, save the runtime train config as a yaml file there,
+        # and the graph construction config, if one exists in the input
         if hasattr(self, "_save_model_path") and self._save_model_path:
-            combined_output_path = os.path.join(self._save_model_path, COMBINED_CONFIG_FILENAME)
-            self._save_combined_config(combined_output_path)
+            # Ensure mode output directory exists
+            os.makedirs(self._save_model_path, exist_ok=True)
+
+            # Save a copy of train config with runtime args
+            train_config_output_path = os.path.join(
+                self._save_model_path, RUNTIME_TRAIN_CONFIG_FILENAME)
+            self._save_runtime_train_config(train_config_output_path)
+
+            # Copy over graph construction config, if one exists
+            gconstruct_config_output_path = os.path.join(
+                self._save_model_path, RUNTIME_GCONSTRUCT_FILENAME)
+            self._copy_graph_construct_config(gconstruct_config_output_path)
+
+    def _copy_graph_construct_config(self, output_data_config):
+        """ Copy graph construct config to the model output path.
+        """
+        # Copy data configuration file if available
+        if get_rank() == 0 and self.part_config is not None:
```

Comment:
`part_config` is a required argument of CLIs. If this PR is for CLIs only, may not need to check `self.part_config` exist or not.

Reply:
There is a check in `_part_config`